### PR TITLE
Add post, update post and delete post

### PR DESCRIPTION
**Pull request by mnath30**
Added the following:
1. User can add the post by clicking on the add icon from the top navigation bar
2. User can edit and delete the existing post from the home page
3. User can see the post by selecting Go to post option from the home page

Added the API calls for adding post, deleting post and updating a post.
Added the API calls for fetching a single post

Note: The single post page is still WIP

### CODE REVIEW

**Review by Kavita539 — APPROVED** [review]
Code looks good and clean. Well maintained code

**Review by SukanyaSen530 — APPROVED** [review]
Modular code.

File: src/Components/Modal/PostUploadModal.jsx [hunk 1]
```diff
@@ -0,0 +1,185 @@
...
+
+  useEffect(() => {
+    if (editingPost) {
+      setPostModalDetails({
+        ...postModalDetails,
+        _id: currentPost._id,
+        imagePreview: currentPost.image,
+        imageInfo: currentPost.description,
+      });
+    }
```
Comment (MarkVed17) [inline src/Components/Modal/PostUploadModal.jsx:33]: May remove this.

File: src/Components/Modal/PostUploadModal.jsx [hunk 2]
```diff
@@ -0,0 +1,185 @@
...
+              <input
+                id="modal-input"
+                type="file"
+                accept="image/png, image/jpg, image/gif, image/jpeg"
+                className="modal__inp"
+                onChange={handleImageSelection}
+              />
+            </label>
+
+            {/* default image when no image is uploaded */}
```
Comment (MarkVed17) [inline src/Components/Modal/PostUploadModal.jsx:104]: Good that you add comments. Makes it easier to understand.

**Review by devananth — APPROVED** [review]
- Well structured code

File: src/Components/PostCards/PostCards.jsx [hunk 3]
```diff
@@ -1,19 +1,49 @@
...
 const PostCards = ({ item, dispatchfunc }) => {
   const { _id, username, profile, info, image, description, likes } = item;
   const { likeCount, likedBy } = likes;
   const encodedToken = localStorage.getItem("encodedToken");
   const currentUserName = localStorage.getItem("username");
   const likedByUser = likedBy.includes(currentUserName);
+  const [showPostDropdown, setShowPostDropdown] = useState(false);
+
+  const handleChange = (e) => {
+    if (!e.currentTarget.contains(e.relatedTarget)) {
```
Comment (MarkVed17) [inline src/Components/PostCards/PostCards.jsx:16]: Oh, I didn’t know about ‘relatedTarget’.

**Review by MarkVed17 — APPROVED** [review]
Liked the naming conventions followed.
Added comments at appropriate places.
You may proceed!

**Pull request merged**